xarray/core/nputils.py

```diff
@@ -206,6 +206,9 @@ def f(values, axis=None, **kwargs):
                 # to ddof=1 above.
                 if pycompat.mod_version("numbagg") < Version("0.7.0"):
                     kwargs.pop("ddof", None)
+                if name == "nanquantile" and kwargs.get("method", "linear") == "linear":
```

Comment:
Sorry — this actually needs to move up into the big `if` expression ending on L197. Because we only want to use `numbagg` if we know we can, otherwise we want to allow bottleneck to work. Does that makes sense? So we should have something like:

```python
# numbagg.nanquantile only available after 0.8.0 and with linear method
and (name != "nanquantile" or (pycompat.mod_version("numbagg") >= Version("0.8.0") and kwargs.get("method", "linear") == "linear"))
```

...above on 197.

...then in this code we can remove ` and kwargs.get("method", "linear") == "linear":` while keeping the kwargs lines below.

Does that make sense? Sorry this ended up being finicky...

Reply:
Makes total sense! Thank you for the support 🙂